internal/ingress/controller/store/store.go

```diff
@@ -296,6 +300,9 @@ func New(
 	store.informers.Ingress = infFactory.Networking().V1().Ingresses().Informer()
 	store.listers.Ingress.Store = store.informers.Ingress.GetStore()
 
+	store.informers.IngressClass = infFactory.Networking().V1().IngressClasses().Informer()
+	store.listers.IngressClass.Store = cache.NewStore(cache.MetaNamespaceKeyFunc)
```

Comment:
Is `cache.NewStore` used here instead of informer's local cache because of thread safety?

Reply:
Nope. I've used this as a New Store, otherwise following the other logics it would fetch everything from the store/informer regardless of the filtering or not.

But can say I've had no idea what I was doing, so I might be wrong here :)